components/micro-gateway-core/src/main/ballerina/src/gateway/filters/pre_authn_filter.bal

```diff
@@ -98,6 +100,34 @@ returns boolean {
                 authHeader = authCookie;
             }
         }
+    } else {
+        //process apikey authentication
+        if (authProvidersIds.indexOf(AUTH_SCHEME_API_KEY) != ()) {
+            json[] apiKeys = getAPIKeysforResource(context.getServiceName(), context.getResourceName());
+            printDebug(KEY_PRE_AUTHN_FILTER, apiKeys.toString());
+            if (apiKeys.length() > 0) {
+                foreach json apiKey in apiKeys {
+                    if (apiKey is  map<json>) {
+                        string inName = apiKey[API_KEY_IN].toString();
+                        string name = apiKey[API_KEY_NAME].toString();
+                        printDebug(KEY_PRE_AUTHN_FILTER, "Detected apikey security in : " + inName + " name: " + name);
+                        if (stringutils:equalsIgnoreCase(HEADER, inName) && request.hasHeader(name)) {
+                            printDebug(KEY_PRE_AUTHN_FILTER, "Request has apikey header : " + name);
+                            isAPIKeyAuth = true;
+                            setAPIKeyAuth(inName, name);
+                            authHeader = AUTH_SCHEME_API_KEY;
+                            break;
+                        } else if (stringutils:equalsIgnoreCase(QUERY, inName) && request.getQueryParamValue(name) is string) {
```

Comment:
120 character limit exceeds.

Reply:
Fixed in https://github.com/wso2/product-microgateway/pull/973